src/editor/manager.ts

```diff
@@ -117,20 +113,29 @@ export default class EditorManager {
     }
   }
 
+  /**
+   * Given an array of Diagnostics remove any duplicates by hashing a key
+   * in the format of from + ' ' + to + ' ' + message.
+   */
+  makeUniqueDiagnostics(duplicatedDiagnostics: Diagnostic[]): Diagnostic[] {
+    const uniqueDiagnostics: Diagnostic[] = []
+    const seenDiagnostic: { [key: string]: boolean } = {}
```

Comment:
Why not use a `Set()`? I'd expect it to be more performant.

Reply:
I didn't use set because the objects are different and the content is the same. So I cannot do a `.has` check because it does not look at the keys, it looks at the object.

Reply:
You could use the string that you made as the value in the set.